server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -245,6 +245,30 @@ public String get(String setting, String defaultValue) {
         return retVal == null ? defaultValue : retVal;
     }
 
+    /**
+     * Returns the setting value associated with the setting key. If it does not exists,
+     * returns the default value provided.
+     */
+    public String get(String setting, String defaultValue, boolean isList) {
+        Object value = settings.get(setting);
+        if(value != null) {
+            if (value instanceof List) {
+                if (isList == false) {
+                    throw new IllegalArgumentException(
+                        "Found list type value for setting [" + setting + "] but but did not expect a list for it."
+                    );
+                }
+            } else if (isList) {
+                throw new IllegalArgumentException(
+                    "Expected list type value for setting [" + setting + "] but found [" + value.getClass() + ']'
```

Comment:
I think that will fail compilation? 😇

Reply:
Compiles just fine for me? :) What am I missing?

Reply:
Oh, never mind, I misread. Sorry for that. 😄